src/components/button/button.tsx

```diff
@@ -57,18 +58,25 @@ export class Button {
     @Prop({ reflect: true })
     public loading = false;
 
-    @Element()
-    private limelButton: HTMLLimelButtonElement;
+    /**
+     * Set to `true` to indicate failure instead of success when the button is
+     * no longer in the `loading` state.
+     */
+    @Prop({ reflect: true })
+    public loadingFailed = false;
+
+    @State()
+    private justLoaded = false;
 
-    private justLoadedTimeout: NodeJS.Timeout;
+    private justLoadedTimeout?: number;
 
     public render() {
         return (
             <button
                 class={`
                     mdc-button
                     ${this.primary ? 'mdc-button--unelevated' : ''}
-                    ${this.loading ? 'loading' : ''}
+                    ${this.renderLoadingState()}
```

Comment:
This function name does not really reflect what it does. It's providing a class and not actually rendering anything

Reply:
What are your atoms of rendering? 😄 Rendering a class name (using the component's state as input) is what that method does.

But I guess you expect a render method to return a rendered element.